### PR TITLE
Feature#2 Generate bussines logic

### PR DESCRIPTION
**Pull request by acorzo**
# Description
Make Branch model
Make Employee model
Make Admin model
Make Attendance model

 # Validations
 - [x] Validate that each branch has a name and the length is greater or equal to 2 characters.
 - [x] Validate that each employee has a name and the length is greater or equal to 2 characters.
 - [x] Validate that each employee has an email address, position, employee_number, private_number.
 - [x]  Validate that employee's email and private_number are unique

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -8,7 +8,9 @@ default: &default
   adapter: postgresql
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
   timeout: 5000
-
+  username: postgres
+  password: 123456
+  
 development:
```
Comment (germanruzca) [inline config/database.yml:14]: Everyone can see your credentials, check this, if the .gitignore is correct. Also, try to create a new branch when the previous is already merged.

Comment (germanruzca) [inline config/database.yml:14]: Everyone can see your credentials, check this, if the .gitignore is correct. Also, try to create a new branch when the previous is already merged.

**Review by germanruzca — CHANGES_REQUESTED** [review]

File: db/seeds.rb [hunk 2]
```diff
@@ -5,3 +5,21 @@
...
+# Employee.create(name: "Maria" ,email: "maria@gmail.com", position: "CTO", employee_number: 4, private_number: 114, active: true, branch_id: 1)	
+# Employee.create(name: "Linda" ,email: "linda@gmail.com", position: "CMO", employee_number: 5, private_number: 115, active: true, branch_id: 1)	
+# Employee.create(name: "Rosa" ,email: "rosa@gmail.com", position: "Manager", employee_number: 6, private_number: 116, active: true, branch_id: 2)	
+# Employee.create(name: "Pedro" ,email: "pedro@gmail.com", position: "Manager", employee_number: 7, private_number: 117, active: true, branch_id: 3)	
+# Employee.create(name: "Zoila" ,email: "zoila@gmail.com", position: "Seller", employee_number: 8, private_number: 118, active: true, branch_id: 4)	
+# Employee.create(name: "Veronica" ,email: "veronica@gmail.com", position: "Seller", employee_number: 9, private_number: 119, active: true, branch_id: 2)	
+# Employee.create(name: "Jose" ,email: "jose@gmail.com", position: "Seller", employee_number: 10, private_number: 120, active: true, branch_id: 4)	
+
+#Admin.create(name:"admin", password:"admin1")
```
Comment (juperameza) [inline db/seeds.rb:25]: Code that you don't need delete it

**Review by juperameza — CHANGES_REQUESTED** [review]

File: config/database.yml (hunk 1 above)
Comment (Jumaquba) [reply config/database.yml]: Right, we have problems with git rebase not bringing in changes from master, where it's ignored.
Your request is on the pending list.

File: db/seeds.rb (hunk 2 above)
Comment (Jumaquba) [reply db/seeds.rb:25]: We create the seeds to test in this branch, we are aware that it should not go to production.
is there another way to do it?

Comment (Jumaquba) [reply db/seeds.rb:25]: we used the faker gem but we wanted the same data for testing.

**Review by germanruzca — DISMISSED** [review]

File: db/seeds.rb (hunk 2 above)
Comment (juperameza) [reply db/seeds.rb:25]: If you will be needing it you should just uncomment it 
Because if someone else needs then they should be able to use it

**Review by juperameza — DISMISSED** [review]

File: db/migrate/20220608204929_create_admins.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class CreateAdmins < ActiveRecord::Migration[7.0]
+  def change
+    create_table :admins do |t|
+      t.string :name
+      t.string :password
+
+      t.timestamps
+    end
+  end
+end
```
Comment (YilKanda) [inline db/migrate/20220608204929_create_admins.rb:10]: Are you planning to use a personal session setup or are you going to use a user setup gem like Devise or Clearence?

File: config/database.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# Warning: The database defined as "test" will be erased and
+# re-generated from your development database when you run "rake".
+# Do not set this db to the same as development or production.
+test:
+  <<: *default
+  database: reloj_db_test
+
+production:
+  <<: *default
+  database: reloj_db_production
```
[the remark this answers is not in the document]
Comment (Jumaquba) [reply config/database.yml]: resolved

File: db/migrate/20220608204929_create_admins.rb (hunk 3 above)
Comment (Jumaquba) [reply db/migrate/20220608204929_create_admins.rb:10]: Authentication from Scratch no gem

File: db/seeds.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jumaquba) [reply db/seeds.rb]: resolved

File: db/seeds.rb [hunk 5]
```diff
@@ -4,4 +4,4 @@
 # Examples:
 #
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
-#   Character.create(name: "Luke", movie: movies.first)
+#   Character.create(name: "Luke", movie: movies.first)
```
[the remark this answers is not in the document]
Comment (acorzo) [reply db/seeds.rb:7]: Hi Yilkanda, This is the example code that comes with ralis by default. No problem.

**Review by V-Juarez — APPROVED** [review]

**Review by AgustinPalmaM — APPROVED** [review]
LGTM

**Review by YilKanda — DISMISSED** [review]
LGTM!

**Review by YilKanda — APPROVED** [review]
LGTM!

**Review by ruruisu — APPROVED** [review]
Looks good

**Review by OwenLobato — APPROVED** [review]

**Pull request merged**